frontend/src/metabase/query_builder/components/view/ViewHeader/components/QuestionDataSource/utils.js

```diff
@@ -81,14 +88,23 @@ export function getDataSourceParts({ question, subHead, isObjectDetail }) {
         }),
     ].filter(isNotNull);
 
-    parts.push(
-      <QuestionTableBadges
-        tables={allTables}
-        subHead={subHead}
-        hasLink={hasTableLink}
-        isLast={!isObjectDetail}
-      />,
-    );
+    const part =
+      formatTableAs === "component" ? (
+        <QuestionTableBadges
+          tables={allTables}
+          subHead={subHead}
+          hasLink={hasTableLink}
+          isLast={!isObjectDetail}
+        />
+      ) : (
+        {
+          name: table.displayName(),
+          href: hasTableLink ? getTableURL(table) : "",
+          model: "table",
+        }
+      );
+
+    parts.push(part);
```

Comment:
Here's the key change. The utility function can either return a whole component that describes a table (which is what it always used to do) or it can return a `Record`.

When we call this function from the `QuestionSources` component in `QuestionInfoSidebar`, we make it return an array of `Record`s.